src/CommonGLPI.php

```diff
@@ -902,6 +902,7 @@ public function showTabsContent($options = [])
 
         if (count($onglets)) {
             $tabpage = $this->getTabsURL();
+            $tabpage = Html::cleanParametersURL($tabpage);
```

Comment:
Instead of stripping params, I think it would be better to preserve them, and to handle them as a `$key => $value` array.

```php
$parsed_url = parse_url($url);
$tab_path = $parsed_url['path'];
$tab_params = [];
if (array_key_exists('query', $parsed_url)) {
    parse_str($parsed_url['query'], $url_params);
}

// TODO Handle $extraparamhtml as a $key => $value array
$tab_params = array_merge($tab_params, $extra_param);

$tab_params = array_merge(
    $tab_params, [
        '_target' => $target,
        '_itemtype' => $this->getType(),
        'id' => $ID,
    ]
);

....

$tabs[$key] = [
    'title'  => $val,
    'url'    => $tabpage,
    'params' => Toolbox::append_params(['_glpi_tab' => $key] + $tab_params, '&amp;'),
];
```

Reply:
It's already in `$extraparamhtml` set before?!

Reply:
The important thing here is that `getTabsURL` is supposed to permit any itemtype to override its tabs URL, including by adding custom query parameters. These parameters should be preserved.
You focused on le last lines of my comment, but the important lines are the first 5 lines.